Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Etch.OrchardCore.Menu.Models;
+using Etch.OrchardCore.Widgets.Models;
+using GraphQL;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Autoroute.Models;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Menu.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Etch.OrchardCore.Menu.Helpers
+{
+    public class CommonMenuItemPartMigrator : ICommonMenuItemPartMigrator
+    {
+        private readonly IContentManager _contentManager;
+        private readonly ILogger<CommonMenuItemPartMigrator> _logger;
+        private readonly ISession _session;
+        private readonly ShellSettings _settings;
+
+        public CommonMenuItemPartMigrator(IContentManager contentManager,
+            ILogger<CommonMenuItemPartMigrator> logger,
+            ISession session,
+            ShellSettings settings)
+        {
+            _contentManager = contentManager;
+            _logger = logger;
+            _session = session;
+            _settings = settings;
+        }
+
+        public async Task GetCommonMenuitemDataAsync()
+        {
+            foreach (var contentItem in await FetchMenuItemsAsync())
+            {
+                var menuItemsListPart = await MenuItemMigratorAsync(contentItem.As<MenuItemsListPart>());
+
+                contentItem.Apply(nameof(MenuItemsListPart), menuItemsListPart);
+                await _contentManager.UpdateAsync(contentItem);
+                await _contentManager.PublishAsync(contentItem);
+            }
+        }
+
+        private async Task<string> ContentItemIdToUrlAsync(string contentItemId)
+        {
+            var path = "/";
+
+            if (!string.IsNullOrEmpty(_settings.RequestUrlPrefix))
+            {
+                path += _settings.RequestUrlPrefix + "/";
+            }
+
+            var contentItem = await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.Latest && x.ContentItemId == contentItemId)
+                .FirstOrDefaultAsync();
+
+            if (contentItem == null)
+            {
+                return path;
+            }
+
+            if (contentItem.Has<AutoroutePart>())
+            {
+                return path += contentItem.As<AutoroutePart>().Path.ToString();
+            }
+
+            return path += $"Contents/ContentItems/{contentItemId}";
+        }
+
+        private async Task<IEnumerable<ContentItem>> FetchMenuItemsAsync()
+        {
+            return await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.Latest && x.ContentType == "Menu")
+                .ListAsync();
+        }
+
+        public async Task<MenuItemsListPart> MenuItemMigratorAsync(MenuItemsListPart menuItemsListPart)

```

Comment:
Doesn't need to be public. Change method name to "MigrateMenuItemsAsync", gives more of an impression of what it's doing to do.